mui/material-ui
# [core] Add React 19 as peer dependency

Includes v19 in React peer dependencies definitions.

**How to test?**

1. Clone this repo: https://github.com/aarongarciah/material-ui-react-19 (it points to v19 React deps and points Material UI deps to the CodeSandbox CI version of the library generated in this PR)
2. Install dependencies: `npm i`
3. Start dev server: `npm start`
4. Build: `npm run build`
5. Check that there's not TypeScript errors.

Follow the same steps with other package managers like `pnpm` and `yarn`.

---

Findings:
- Installing dependencies using `npm` (teted in Node v18 and v20) results in an error with an `@emotion/react` peer dep. To overcome the error, you need to use the `--force` flag (`--legacy-peer-deps` doesn't work).
  <img width="872" alt="Screenshot 2024-08-07 at 14 14 16" src="https://github.com/user-attachments/assets/9eb965db-b3a6-4d92-92a7-27f93b96659b">
- `pnpm` and `yarn` (v1) seem to work well
- No TypeScript errors reported.
